app/controlplane/plugins/sdk/v1/fanout.go

```diff
@@ -258,6 +263,14 @@ func (i AvailablePlugins) FindByID(id string) (FanOut, error) {
 	return nil, fmt.Errorf("integration %q not found", id)
 }
 
+func (i AvailablePlugins) Cleanup() {
+	for _, plugin := range i {
+		if plugin.DisposeFunc != nil {
+			plugin.DisposeFunc()
```

Comment:
Question, this `DisposeFunc` is implemented by the plugin itself right? I'm think [here](https://github.com/chainloop-dev/chainloop/pull/207/files#diff-40bd4cdc92f65dfa360f7c52d54d44a7fa3f96b373b3f3b18ecfc90b8fe0eda8R116), when chain loop control plane is shutting down it will call every `CleanUp` func of the plugins.

What if the plugin has unfinished data/process to be performed but the core is shutting down? Is there a process of gracefully stop them and wait for a signal to actually stop? That's my worry, plugins with unfashioned business losing its process because of a shutdown.

If it relays on the plugin's developer decision, I would state it somehow so it's clear on the consequences if, the plugin manages the concept of states of course.

Reply:
Actually not, dispose function is a function injected by the initialization process in the host, which in the case of go-plugins plugins means killing the connection 

```go
return &sdk.FanOutP{FanOut: fanOut, DisposeFunc: client.Kill}, nil
```

but your point is in fact how it should evolve indeed, a graceful exit mechanism, I just think we are not there yet. I'll create a task to keep it in mind for future iterations of the SDK.

Thanks!